go/oasis-node/cmd/common/metrics/config/config.go

```diff
@@ -0,0 +1,61 @@
+// Package config implements global metrics configuration options.
+package config
+
+import (
+	"fmt"
+	"time"
+)
+
+// Config is the metrics configuration structure.
+type Config struct {
+	// Metrics mode (none, pull, push).
+	Mode string `yaml:"mode"`
+	// Metrics pull address.
+	Address string `yaml:"address"`
+
+	// Metrics push job name (debug-only).
+	JobName string `yaml:"job_name,omitempty"`
+	// Metrics push instance labels (debug-only).
+	Labels map[string]string `yaml:"labels,omitempty"`
+	// Metrics push interval (debug-only).
+	Interval time.Duration `yaml:"interval,omitempty"`
+}
+
+// Validate validates the configuration settings.
+func (c *Config) Validate() error {
+	switch c.Mode {
+	case "none":
+	case "pull":
+		if len(c.Address) == 0 {
```

Comment:
Could use `_, err = net.ResolveTCPAddr` or something similar, if that is not too strict.

Reply:
The address could also be a UNIX socket or something else and the machine that's running the migration command might not be the machine that will actually be running the node :)

I don't want to force a specific type of address here -- the validation for this can happen once the code actually tries to connect to the address.  This is also similar to what's done in other config modules, the really basic stuff can be verified in the config module, but anything more complex should be done in the code that uses it instead.